lib/engine/ability/exchange.rb

```diff
@@ -8,7 +8,7 @@ class Exchange < Base
       attr_reader :corporation, :from
 
       def setup(corporation:, from:)
-        @corporation = corporation
+        @corporation = corporation == 'any' ? corporation.to_sym : corporation
```

Comment:
keep the types the same, so always do sym or always expect str

Reply:
This is where the json string is converted to a symbol. It's used as a symbol throughout. Was there a different place you were thinking for the conversion?

Reply:
i'd just leave at as @corporation = corporation.to_sym then

Reply:
Using a symbol will break individual certificates. I went ahead and used a string for any, so the type is the same regardless of value.